features/upgrade/sdn/policy-upgrade.feature

```diff
@@ -477,3 +477,180 @@ Feature: SDN compoment upgrade testing
+    Given I use the "<%= cb.proj2 %>" project
+    Given I obtain test data file "networking/list_for_pods.json"
+    When I run oc create over "list_for_pods.json" replacing paths:
+      | ["items"][0]["spec"]["replicas"] | 1 |
+    Then the step should succeed
+    Given a pod becomes ready with labels:
+      | name=test-pods |
+    And evaluation of `pod(2).name` is stored in the :p2pod1name clipboard
+    Given I obtain test data file "rc/idle-rc-1.yaml"
+    When I run oc create over "idle-rc-1.yaml" replacing paths:
+      | ["items"][0]["spec"]["replicas"] | 1 |
+    Then the step should succeed
+    Given a pod becomes ready with labels:
+      | name=hello-idle |
+    And evaluation of `pod(3).name` is stored in the :p2pod2name clipboard
+    When I execute on the "<%= cb.p2pod1name %>" pod:
+      | curl | -I | <%= cb.p1pod1ip %>:8080 | --connect-timeout | 5 | 
+    Then the output should contain "200 OK"
+    Then the step should succeed
+    Given I use the "<%= cb.proj0 %>" project
+    And I wait up to 20 seconds for the steps to pass:
+    """
+    When I execute on the "<%= cb.rsyslog_server_pod %>" pod:
+      | grep | -w | verdict=allow | /var/log/messages |
+    Then the output should contain "verdict=allow"
+    """
+    
+    Given I use the "<%= cb.proj2 %>" project
+    When I execute on the "<%= cb.p2pod2name %>" pod:
+      | curl | -I | <%= cb.p1pod1ip %>:8080 | --connect-timeout | 20 |
+    Then the output should not contain "200 OK"
+    And I wait up to 20 seconds for the steps to pass:
+    """
+    Then the step should fail
+    Given I use the "<%= cb.proj0 %>" project
+    When I execute on the "<%= cb.rsyslog_server_pod %>" pod:
+      | grep | -w | verdict=drop | /var/log/messages |
+    Then the output should contain "verdict=drop"
+    """
+
+  # @author asood@redhat.com
+  # @case_id OCP-44978
+  @admin
+  @upgrade-check
+  @4.10 @4.9
+  @aws-ipi @gcp-ipi @vsphere-ipi @azure-ipi @baremetal-ipi @openstack-ipi
+  @aws-upi @gcp-upi @openstack-upi
+  @destructive
+  @network-ovnkubernetes
+  Scenario: Check network policy ACL logging works post upgrade 
+    Given I switch to cluster admin pseudo user
+    Given the env is using "OVNKubernetes" networkType
+    And evaluation of `"rsyslog-server"` is stored in the :proj0 clipboard
+    And evaluation of `"policy-acl-upgrade1"` is stored in the :proj1 clipboard
+    And evaluation of `"policy-acl-upgrade2"` is stored in the :proj2 clipboard
+    Given I use the "<%= cb.proj0 %>" project
+    Given I wait for the "rsyslogserver" service to become ready
+    Given a pod becomes ready with labels:
+      | appname=rsyslogserver |
+    And evaluation of `service("rsyslogserver").ip` is stored in the :svc_ip clipboard
+    And evaluation of `pod(0).name` is stored in the :rsyslog_server_pod clipboard
+    Then the step should succeed
+
+    Given as admin I successfully merge patch resource "networks.operator.openshift.io/cluster" with:
```

Comment:
do we still need this patch since it already be done before upgrade?

Reply:
Yes, because post upgrade syslog server pod is recreated with different IP address.

Reply:
Thanks @zhaozhanqi for the review. 
Fixed it.

Reply:
Thanks for updating this. it looks good to me